confidential-data-hub/hub/src/api.rs

```diff
@@ -26,4 +27,6 @@ pub trait DataHub {
     /// URI is defined in
     /// <https://github.com/confidential-containers/guest-components/blob/main/attestation-agent/docs/KBS_URI.md>
     async fn get_resource(&self, uri: String) -> Result<Vec<u8>>;
+
+    async fn secure_mount(&self, storage: Storage) -> Result<String>;
```

Comment:
it is suggested that the parameter `storage` we use `str` or `&[u8]`. And we do the deserialization inside the impl of the function.

In this way we can avoid import `storage` in outer files like `api.rs` and `server.rs`.